packages/frontend/src/components/composer/Composer.tsx

```diff
@@ -129,6 +132,25 @@ const Composer = forwardRef<
     ? editMessageInputRef
     : regularMessageInputRef
 
+  const saveVoiceAsDraft = (voiceData: Blob) => {
+    const reader = new FileReader()
+    reader.readAsDataURL(voiceData)
+    reader.onloadend = async () => {
+      if (!reader.result) {
+        log.error('Cannot convert blob to base64. reader.result is null')
+        return
+      }
+      const b64 = reader.result.toString().split(',')[1]
+      const filename = b64.slice(3, 12).toString() + '.mp3'
+      const path = await runtime.writeTempFileFromBase64(filename, b64)
+      currentComposerMessageInputRef.current?.setState({ loadingDraft: false })
```

Comment:
This should not be necessary but at the time saveVoiceAsDraft is called the currentComposerMessageInputRef.state is still on "loadingDraft: true" and addFileToDraft fails ?